src/applications/personalization/dashboard/components/health-care-v2/HealthCareContentV2.jsx

```diff
@@ -0,0 +1,325 @@
+                  onClick={() =>
+                    recordEvent({
+                      event: 'nav-linkslist',
+                      'links-list-header': 'Review your messages',
+                      'links-list-section-header': 'Health care',
+                    })
+                  }
+                />
+              </div>
+            </va-alert>
+          </div>
+        ) : null}
+        {hasAppointmentsError && <HealthcareError />}
+        {hasUpcomingAppointment && (
+          /* Appointments */
+          <Appointments appointments={appointments} />
+        )}
+        {!isVAPatient && <NoHealthcareText />}
+        {isVAPatient &&
+          !hasUpcomingAppointment &&
+          !hasAppointmentsError && <NoUpcomingAppointmentsText />}
+        {shouldShowOnOneColumn && !hasAppointmentsError ? (
+          <HealthCareCTA
+            hasInboxError={hasInboxError}
+            authenticatedWithSSOe={authenticatedWithSSOe}
+            hasUpcomingAppointment={hasUpcomingAppointment}
+            shouldShowPrescriptions={shouldShowPrescriptions}
+            unreadMessagesCount={unreadMessagesCount}
+            isVAPatient={isVAPatient}
+          />
+        ) : null}
+      </DashboardWidgetWrapper>
+      {!shouldShowOnOneColumn && !hasAppointmentsError ? (
+        <DashboardWidgetWrapper>
+          <HealthCareCTA
+            hasInboxError={hasInboxError}
+            authenticatedWithSSOe={authenticatedWithSSOe}
+            hasUpcomingAppointment={hasUpcomingAppointment}
+            shouldShowPrescriptions={shouldShowPrescriptions}
+            unreadMessagesCount={unreadMessagesCount}
+            isVAPatient={isVAPatient}
+          />
+        </DashboardWidgetWrapper>
+      ) : null}
+    </div>
+  );
+};
+
+const mapStateToProps = state => {
+  const facilityLocations = [
+    'VA Spokane health care',
+    'VA Walla Walla health care',
+    'VA Central Ohio health care',
+    'Roseburg (Oregon) VA health care',
+    'White City health care',
+  ];
+
+  const shouldFetchUnreadMessages = selectAvailableServices(state).includes(
+    backendServices.MESSAGING,
+  );
+  const shouldShowPrescriptions = selectAvailableServices(state).includes(
+    backendServices.RX,
+  );
+
+  const fetchingAppointments = state.health?.appointments?.fetching;
+  const fetchingUnreadMessages = shouldFetchUnreadMessages
+    ? selectUnreadCount(state)?.fetching
+    : false;
+
+  const hasUnreadMessagesCountError =
+    selectUnreadCount(state)?.errors?.length > 0;
+  const hasAppointmentsError = state.health?.appointments?.errors?.length > 0;
+
+  return {
+    appointments: state.health?.appointments?.data,
+    authenticatedWithSSOe: isAuthenticatedWithSSOe(state),
+    facilityLocations,
+    hasInboxError: hasUnreadMessagesCountError,
+    hasAppointmentsError,
+    isCernerPatient: selectIsCernerPatient(state),
+    shouldFetchUnreadMessages,
+    // TODO: We might want to rewrite this component so that we default to
```

Comment:
I agree that we should re-evaluate the data fetching process in our apps. i really wish we could use React Router 6.4+ and use their loader pattern, which is just like Remix's loaders.

I'll think about that and see if maybe we could collaborate on getting this data fetched as early as possible to avoid layout janks and shift.